source/binding/Statements.cpp

```diff
@@ -686,6 +709,10 @@ bool VariableDeclStatement::verifyConstantImpl(EvalContext& context) const {
     return true;
 }
 
+void VariableDeclStatement::serializeTo(ASTSerializer& serializer) const {
+    serializer.write("symbol", symbol);
```

Comment:
I think you want to use writeLink() here instead, because this VariableDeclStatement is referencing a variable that is owned by the block symbol. The VariableDeclStatement is a way of saying "this variable gets initialized at this point relative to other statements in the block". Using write() instead of writeLink() will cause the full definition of the symbol to show up in the output twice.